impl_core/src/test/java/io/opencensus/implcore/trace/export/SpanExporterImplTest.java

```diff
@@ -170,8 +194,40 @@ public void exportSpansToMultipleServices() {
 
   @Test
   public void exportNotSampledSpans() {
-    SpanImpl span1 = createNotSampledEndedSpan(SPAN_NAME_1);
-    SpanImpl span2 = createSampledEndedSpan(SPAN_NAME_2);
+    SpanExporterImpl spanExporter = SpanExporterImpl.create(4, Duration.create(1, 0));
+    StartEndHandler startEndHandler =
+        new StartEndHandlerImpl(spanExporter, runningSpanStore, null, new SimpleEventQueue());
+
+    spanExporter.registerHandler("test.service", serviceHandler);
+
+    SpanImpl span1 = createNotSampledEndedSpan(startEndHandler, SPAN_NAME_1);
+    SpanImpl span2 = createSampledEndedSpan(startEndHandler, SPAN_NAME_2);
+    // Spans are recorded and exported in the same order as they are ended, we test that a non
+    // sampled span is not exported by creating and ending a sampled span after a non sampled span
+    // and checking that the first exported span is the sampled span (the non sampled did not get
+    // exported).
+    List<SpanData> exported = serviceHandler.waitForExport(1);
+    // Need to check this because otherwise the variable span1 is unused, other option is to not
+    // have a span1 variable.
+    assertThat(exported).doesNotContain(span1.toSpanData());
+    assertThat(exported).containsExactly(span2.toSpanData());
+  }
+
+  @Test
+  public void exportNotSampledSpansFlushed() {
+    // Set the export delay to a very long value in order to confirm the #flush() below works
+    SpanExporterImpl spanExporter = SpanExporterImpl.create(4, Duration.create(Long.MAX_VALUE, 0));
```

Comment:
It looks like `Duration.create` has a limit on the number of seconds, and it uses zero when the value is greater than 315576000000L:

https://github.com/census-instrumentation/opencensus-java/blob/773c7638f448bdb6d33d1c30bc8ec4ba5450da18/api/src/main/java/io/opencensus/common/Duration.java#L55-L56

I created an issue to improve the API: #1179